apl-core/src/main/java/com/apollocurrency/aplwallet/apl/core/rest/endpoint/DexController.java

```diff
@@ -603,11 +605,82 @@ public Response getUserFilledOrders(@Parameter(description = "Number of first N
             responses = @ApiResponse(description = "List of versioned contracts", responseCode = "200",
                     content = @Content(mediaType = "application/json", schema = @Schema(implementation = ExchangeContractDTO.class))))
     public Response getAllVersionedContractsForOrder(@Parameter(description = "APL account id (RS, singed or unsigned int64/long) ") @QueryParam("accountId") String account,
-                                        @Parameter(description = "Order id (signed/unsigned int64/long) ") @QueryParam("orderId") String order) {
+                                                     @Parameter(description = "Order id (signed/unsigned int64/long) ") @QueryParam("orderId") String order) {
         long accountId = Convert.parseAccountId(account);
         long orderId = Convert.parseLong(order);
         List<ExchangeContract> contracts = service.getVersionedContractsByAccountOrder(accountId, orderId);
         return Response.ok(contractConverter.convert(contracts)).build();
     }
 
+    @GET
+    @Path("/eth/addresses")
+    @Produces(MediaType.APPLICATION_JSON)
+    @Operation(tags = {"dex"}, description = "Get all user addresses on the smart contract. ",
+        responses = @ApiResponse(description = "List of user addresses", responseCode = "200",
+            content = @Content(mediaType = "application/json", schema = @Schema(implementation = String.class))))
+    public Response getAllUsers() {
+        List<String> addresses;
+        try {
+            addresses = service.getAllUsers();
+        } catch (AplException.ExecutiveProcessException e) {
+            return Response.status(Response.Status.BAD_REQUEST).build();
```

Comment:
Request is OK, server encountered with unexpected error, so that better to return 500 code here.